geomesa-raster/src/main/scala/org/locationtech/geomesa/raster/data/AccumuloRasterStore.scala

```diff
@@ -0,0 +1,366 @@
+    val resultingBounds = SelfClosingScanner(scanner)
+    if (resultingBounds.isEmpty) {
+      BoundingBox(-180, 180, -90, 90)
+    } else {
+      //TODO: GEOMESA-646 anti-meridian questions
+      reduceValuesToBoundingBox(resultingBounds.map(_.getValue))
+    }
+  }
+
+  // TODO: Consider adding resolutions + extent info  https://geomesa.atlassian.net/browse/GEOMESA-645
+  def getAvailableResolutions: Seq[Double] = getAvailabilityMap.keySet.toSeq.sorted
+
+  def getAvailableGeoHashLengths: Set[Int] = getAvailabilityMap.values().toSet
+
+  def getAvailabilityMap: ImmutableSetMultimap[Double, Int] = AccumuloRasterStore.boundsCache.get(tableName, callable)
+
+  def callable = new Callable[ImmutableSetMultimap[Double, Int]] {
+    override def call(): ImmutableSetMultimap[Double, Int] = {
+      ensureBoundsTableExists()
+      val scanner = connector.createScanner(GEOMESA_RASTER_BOUNDS_TABLE, getAuths)
+      scanner.setRange(new Range(getBoundsRowID))
+      val scanResultingKeys = SelfClosingScanner(scanner).map(_.getKey).toSeq
+      val geohashlens = scanResultingKeys.map(_.getColumnFamily.toString).map(lexiDecodeStringToInt)
+      val resolutions = scanResultingKeys.map(_.getColumnQualifier.toString).map(lexiDecodeStringToDouble)
+      val m = new ImmutableSetMultimap.Builder[Double, Int]()
+      (resolutions zip geohashlens).foreach(x => m.put(x._1, x._2))
+      m.build()
+    }
+  }
+
+  def getGridRange: GridEnvelope2D = {
+    val bounds = getBounds
+    val resolutions = getAvailableResolutions
+    // If no resolutions are available, then we have an empty table so assume 1.0 for now
+    val resolution = resolutions match {
+      case Nil => 1.0
+      case _   => resolutions.min
+    }
+    val width  = Math.abs(bounds.getWidth / resolution).toInt
+    val height = Math.abs(bounds.getHeight / resolution).toInt
+
+    new GridEnvelope2D(0, 0, width, height)
+  }
+
+  def adaptIteratorToChunks(iter: java.util.Iterator[Entry[Key, Value]]): Iterator[Raster] = {
+    iter.map(entry => schema.decode((entry.getKey, entry.getValue)))
+  }
+
+  private def dateToAccTimestamp(dt: DateTime): Long =  dt.getMillis / 1000
+
+  private def createBoundsMutation(raster: Raster): Mutation = {
+    // write the bounds mutation
+    val mutation = new Mutation(getBoundsRowID)
+    val value = bboxToValue(BoundingBox(raster.metadata.geom.getEnvelopeInternal))
+    val resolution = lexiEncodeDoubleToString(raster.resolution)
+    val geohashlen = lexiEncodeIntToString(raster.minimumBoundingGeoHash.map( _.hash.length ).getOrElse(0))
+    mutation.put(geohashlen, resolution, value)
+    mutation
+  }
+
+  private def createMutation(raster: Raster): Mutation = {
+    val (key, value) = schema.encode(raster, writeVisibilities)
+    val mutation = new Mutation(key.getRow)
+    val colFam   = key.getColumnFamily
+    val colQual  = key.getColumnQualifier
+    val colVis   = key.getColumnVisibilityParsed
+    // TODO: WCS: determine if this is wise/useful
```

Comment:
GEOMESA-562 is marked as a Done/Won't Fix. If this is indeed the case, please remove the TODO.